### PR TITLE
Closes #1529 : Cardify the Ingredient tab

### LINKED ISSUE #1529: Cardify the Ingredient tab
Cardify the Ingredient tab, just like the summary tab.

### PR DESCRIPTION
**Pull request by jaindiv26**
## Description
Cardified the Ingredient tab, just like the summary tab.

## Related issues and discussion
Cardify the Ingredient tab
#1529 
 
 ## Screen-shots, if any
screenshot_1525715733
screenshot_1525715730

### CODE REVIEW

**Comment by PrajwalM2212** [comment]
From your current implementation other aspects such as additives  and allergens will appear in one card. 
Probably seperate cards for each aspect would be nice . Ping @Karljoones @huzaifaiftikhar

**Comment by PrajwalM2212** [comment]
@jaindiv26 wait till other reviewers give their opinions before making any chnage.

**Comment by jaindiv26** [comment]
Ohk, I'll wait for other reviewers.

**Comment by huzaifaiftikhar** [comment]
I think separate cards will look nice rather than clubbing all the fields in a single card.

**Comment by Karljoones** [comment]
I personally like the idea of them all being in one card, I think that if you have multiple cards it can look quite overcrowded. However, I would say that I think some of the padding between elements could be reduced

**Comment by huzaifaiftikhar** [comment]
@Karljoones I guess you are talking about the padding to be reduced in the Overview tab? For this tab a single card looks nice and reducing padding between the elements will make it look nicer. However, for the Ingredients tab we can have seperate card for Ingredients List, Substances causing allergies, Additives, Ingredients from Palm oil etc. What do you think about this? :slightly_smiling_face:

**Comment by Karljoones** [comment]
@huzaifaiftikhar Yes, sorry for not being clearer! As you said yourself, the overview tab should be one card with some of the padding removed. 

And yes, I also agree with your second point. There should only be cards when necessary to make sections of the information different. Especially if we are looking to inform users, bunching these together would not make sense. People are going to get bored half way through reading a long list of ingredients, so putting information they need separately would be beneficial.

**Comment by jaindiv26** [comment]
So, I've changed the padding of overview tab and earlier it was looking odd as the padding and margin aren't equal, so I've made them equal. And @huzaifaiftikhar, @Karljoones  I think both the tabs should look similar to have a perfect symmetry, I mean to say having all the fields in one card view only with similar attributes like font size, padding, margin, etc. What do you think about this?

Here are the changes that I've made in the overview tab.

screenshot_1525793662

**Comment by PrajwalM2212** [comment]
@jaindiv26  We have collectively decided that having different cards for different  fields is the way to go .
@Karljoones has clearly explained that long lists are boring to read.

**Comment by jaindiv26** [comment]
Ingredient, Substance, Trace, Additive, Palm Oil Product, MaybeFromPalmOilProduct, vitaminsTagsText, mineralTagsText, aminoAcidTagsText, otherNutrientsTagsText. Currently, these 10 textViews are present in the XML file, Can you please tell me @PrajwalM2212 which textView should be covered with CardView.

**Comment by PrajwalM2212** [comment]
@jaindiv26 If you read over the previous comments it will be clear that all fields that make individual sense can be in a separate card view.

**Comment by jaindiv26** [comment]
@PrajwalM2212 Please check the changes I've made in both the files.

File: app/src/main/res/layout/fragment_ingredients_product.xml [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    android:id="@+id/swipeRefresh"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    app:layout_behavior="@string/appbar_scrolling_view_behavior">
+                                              xmlns:app="http://schemas.android.com/apk/res-auto"
+                                              android:id="@+id/swipeRefresh"
+                                              android:layout_width="match_parent"
+                                              android:layout_height="match_parent"
+                                              app:layout_behavior="@string/appbar_scrolling_view_behavior">
```
Comment (huzaifaiftikhar) [inline app/src/main/res/layout/fragment_ingredients_product.xml]: Remove extra spaces from these lines. Reformat code for proper indentation.

**Comment by huzaifaiftikhar** [comment]
@jaindiv26 switch to open food facts from build variants and then provide some screenshots of the changes you have made.

**Comment by jaindiv26** [comment]
@huzaifaiftikhar I've cardified Substance, Trace, Additive, Palm Oil Product, MaybeFromPalmOilProduct, vitaminsTagsText, mineralTagsText, aminoAcidTagsText, otherNutrientsTagsText just like Ingredients field. As I'm unable to find a product with all these fields, I can assume that these fields have been cardified just like Ingredients field. 

Here's how Ingredients filed looks like after being cardified. 

screenshot_1525870869

**Comment by jaindiv26** [comment]
@huzaifaiftikhar App is unable to show search results for "Classic Burger". Although I can see search results on main website.

**Comment by huzaifaiftikhar** [comment]
@jaindiv26 Have you switched to Open Food Facts ?

**Comment by PrajwalM2212** [comment]
@jaindiv26 Please refer the image under installation section https://github.com/openfoodfacts/openfoodfacts-androidapp

**Comment by jaindiv26** [comment]
@huzaifaiftikhar Sorry I forgot to change the build variants, I found the search result and here is the screenshot. 

screenshot_1525873325
screenshot_1525873332

**Comment by PrajwalM2212** [comment]
The padding between the cards seems to be too much.

**Comment by jaindiv26** [comment]
@PrajwalM2212 Currently the cardview layout_margin is at 16 dp.

**Comment by PrajwalM2212** [comment]
Yes .. that seems to be the problem . You may want to consider reducing it to 8dp or any other lesser number till it appears good.

**Comment by huzaifaiftikhar** [comment]
I have approved previous changes. Consider reducing the padding between the cards.

**Comment by jaindiv26** [comment]
@PrajwalM2212 How about now? To me it looks good, tell me about any other changes to be made.
screenshot_1525875053
screenshot_1525875057

**Comment by jaindiv26** [comment]
@huzaifaiftikhar and @PrajwalM2212 Please see the changes in the above screenshot, it looks quite good now, as I've managed to equal the margin and padding of above layout.

**Comment by jaindiv26** [comment]
Please, someone, help me with my build so this issue can be closed. Travis seems to be showing errors.

**Comment by ross-holloway94** [comment]
@jaindiv26 Travis were having a central issue. I've restarted your build, as it just passed my PR.

**Comment by huzaifaiftikhar** [comment]
Merged, thanks @jaindiv26

**Comment by jaindiv26** [comment]
@huzaifaiftikhar @PrajwalM2212 @Karljoones Thanks for guiding me.

**Pull request merged**